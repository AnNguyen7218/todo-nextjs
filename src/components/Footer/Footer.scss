@import 'config.colors';
@import 'config.gutters';
@import 'config.fonts';
@import 'config.variables';

.auto-hidden {
  display: none;
  @include break(lg) {
    display: block;
  }
}

.wrapper {
  width: 100vw;
  background-color: $black;
  padding-top: gutter(20);
  padding-bottom: gutter(16);
  height: $footer-height-sm;

  @include break(md) {
    height: $footer-height-md;
  }

  @include break(lg) {
    padding-top: gutter(48);
    padding-bottom: gutter(48);
    height: $footer-height-lg;
  }

  @include break(xl) {
    height: $footer-height-xl;
  }
}

.sub-title {
  color: $white;
  opacity: 0.5;
}

.top-section {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-bottom: gutter(12);
  @include break(lg) {
    padding-bottom: gutter(16);
  }
}

.bottom-section {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-top: gutter(12);
  border-top-width: 0.05rem;
  border-top-color: #4d4d4d;
  border-top-style: solid;

  @include break(lg) {
    padding-top: gutter(16);
  }
}

.texts-wrapper,
.icons-wrapper {
  display: flex;
  align-items: center;
}

.link {
  margin-left: gutter(12);
  @include break(lg) {
    margin-left: gutter(16);
  }
}

.link-text {
  @include small;
}

.icon {
  height: 1.5rem;
  width: 1.5rem;

  @include break(lg) {
    width: 1.8rem;
    height: 1.8rem;
  }
}
