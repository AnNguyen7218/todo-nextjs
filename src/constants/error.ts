export const ERROR_MESSAGE = {
  SERVER_ERROR: 'Server error, please try again.',
  REQUIRED: '*Required field',
  INVALID_EMAIL: 'Invalid email address',
  TAKEN_EMAIL: 'Email already exists',
  PASSWORD_NO_MATCH: "Password doesn't match",
  INVALID_PASSWORD:
    'Must contain at least 8 characters with upper and lower case and numbers',
  USER_NOT_EXIST:
    "Sorry, we couldn't find an account with that email. Please enter another email or sign up for a new account.",
};

const AUTH_FORM_ERROR_FIRSTNAME_PROVIDE = 'Auth.form.error.firstName.provide';
const AUTH_FORM_ERROR_LASTNAME_PROVIDE = 'Auth.form.error.lastName.provide';
const AUTH_FORM_ERROR_EMAIL_PROVIDE = 'Auth.form.error.email.provide';
const AUTH_FORM_ERROR_EMAIL_TAKEN = 'Auth.form.error.email.taken';
const AUTH_FORM_ERROR_EMAIL_FORMAT = 'Auth.form.error.email.format';
const AUTH_FORM_ERROR_PASSWORD_PROVIDE = 'Auth.form.error.password.provide';
const AUTH_FORM_ERROR_PASSWORD_FORMAT = 'Auth.form.error.password.format';
const AUTH_FORM_ERROR_USER_NOT_EXIST = 'Auth.form.error.user.not-exist';
const AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE =
  'Auth.form.error.passwordConfirmation.provide';
const AUTH_FORM_ERROR_CODE_PROVIDE = 'Auth.form.error.code.provide';
const AUTH_FORM_ERROR_PASSWORD_MATCHING = 'Auth.form.error.password.matching';
const AUTH_FORM_ERROR_PARAMS_PROVIDE = 'Auth.form.error.params.provide';

export type ServerErrorID =
  | typeof AUTH_FORM_ERROR_FIRSTNAME_PROVIDE
  | typeof AUTH_FORM_ERROR_LASTNAME_PROVIDE
  | typeof AUTH_FORM_ERROR_EMAIL_PROVIDE
  | typeof AUTH_FORM_ERROR_EMAIL_TAKEN
  | typeof AUTH_FORM_ERROR_EMAIL_FORMAT
  | typeof AUTH_FORM_ERROR_PASSWORD_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE
  | typeof AUTH_FORM_ERROR_CODE_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_MATCHING
  | typeof AUTH_FORM_ERROR_PARAMS_PROVIDE
  | typeof AUTH_FORM_ERROR_PASSWORD_FORMAT;

export const SERVER_ERROR_MAP = {
  [AUTH_FORM_ERROR_FIRSTNAME_PROVIDE]: {
    field: 'firstName',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_LASTNAME_PROVIDE]: {
    field: 'lastName',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_EMAIL_PROVIDE]: {
    field: 'email',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_EMAIL_TAKEN]: {
    field: 'email',
    message: ERROR_MESSAGE.TAKEN_EMAIL,
  },
  [AUTH_FORM_ERROR_EMAIL_FORMAT]: {
    field: 'email',
    message: ERROR_MESSAGE.INVALID_EMAIL,
  },
  [AUTH_FORM_ERROR_PASSWORD_PROVIDE]: {
    field: 'password',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_PASSWORD_FORMAT]: {
    field: 'password',
    message: ERROR_MESSAGE.INVALID_PASSWORD,
  },
  [AUTH_FORM_ERROR_USER_NOT_EXIST]: {
    field: 'form', // will show on top of form
    message: ERROR_MESSAGE.USER_NOT_EXIST,
  },
  [AUTH_FORM_ERROR_PASSWORD_CONFIRMATION_PROVIDE]: {
    field: 'passwordConfirmation',
    message: ERROR_MESSAGE.REQUIRED,
  },
  [AUTH_FORM_ERROR_PASSWORD_MATCHING]: {
    field: 'passwordConfirmation',
    message: ERROR_MESSAGE.PASSWORD_NO_MATCH,
  },
  [AUTH_FORM_ERROR_CODE_PROVIDE]: {
    field: 'form',
    message: '',
  },
  [AUTH_FORM_ERROR_PARAMS_PROVIDE]: {
    field: 'form',
    message: '',
  },
};
