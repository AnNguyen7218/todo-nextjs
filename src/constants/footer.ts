import FB from 'assets/icons/ic-facebook.svg';
import Insta from 'assets/icons/ic-insta.svg';
import Twitter from 'assets/icons/ic-twitter.svg';
import LinkIn from 'assets/icons/ic-linkin.svg';

export const FOOTER_LINKS_ICON: FooterLinkIcon[] = [
  {
    url: '/#',
    icon: FB,
  },
  {
    url: '/#',
    icon: Insta,
  },
  {
    url: '/#',
    icon: Twitter,
  },
  {
    url: '/#',
    icon: LinkIn,
  },
];

export const FOOTER_LINKS_TEXT: FooterLinkText[] = [
  {
    url: '/#',
    title: 'Term',
  },
  {
    url: '/#',
    title: 'Privacy',
  },
];
