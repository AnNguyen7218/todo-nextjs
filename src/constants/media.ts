export const MEDIA_MD_UP = '(min-width: 720px)';
export const MEDIA_LG_UP = '(min-width: 960px)';

export const LG = 960;
export const XL = 1200;
