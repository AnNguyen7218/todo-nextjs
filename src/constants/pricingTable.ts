import {
  PlanKey,
  ServiceRowOrder,
  PlanHeaders,
  PlanServices,
  PlansOption,
  Services,
  MobileService,
} from 'types/pricingTable';
import Checked from 'assets/icons/ic-checked.svg';
import StartUpIcon from 'assets/icons/ic-startup.svg';
import ProfessionalIcon from 'assets/icons/ic-professional.svg';
import StudioIcon from 'assets/icons/ic-studio.svg';
import GrowthIcon from 'assets/icons/ic-rocket.svg';

// plans
export const STARTUP = 'STARTUP';
export const PROFESSIONAL = 'PROFESSIONAL';
export const STUDIO = 'STUDIO';
export const GROWTH = 'GROWTH';

// services
export const SERVICES_SHOWCASE = 'SERVICES_SHOWCASE';
export const SERVICE_LISTING = 'SERVICE_LISTING';
export const VERIFIED_PROFILE = 'VERIFIED_PROFILE';
export const FEATURED_PROFILE = 'FEATURED_PROFILE';
export const BUSINESS_DEVELOPMENT = 'BUSINESS_DEVELOPMENT';
export const AUTHENTICATION_VERIFICATION_SUPPORT =
  'AUTHENTICATION_VERIFICATION_SUPPORT';
export const VERIFIED_CLIENT_REVIEWS = 'VERIFIED_CLIENT_REVIEWS';
export const CLIENT_REVIEW_WITH_FOLLOW_UP = 'CLIENT_REVIEW_WITH_FOLLOW_UP';
export const RECEIVE_RESPOND_INQUIRIES = 'RECEIVE_RESPOND_INQUIRIES';
export const SUB_ACCOUNTS = 'SUB_ACCOUNTS';
export const CUSTOMER_SERVICE = 'CUSTOMER_SERVICE';
export const PERSONALISED_CUSTOMER_SERVICE = 'PERSONALISED_CUSTOMER_SERVICE';
export const LEGAL_ADVISORY_SERVICE = 'LEGAL_ADVISORY_SERVICE';
export const PRODUCT_PROFILE_ADVISORY_SERVICE =
  'PRODUCT_PROFILE_ADVISORY_SERVICE';

// data type
export const TEXT = 'TEXT';
export const ICON = 'ICON';

// row theme
export const WHITE = 'white';
export const GRAY = 'gray';

export const PLANS_ORDER: PlanKey[] = [STARTUP, PROFESSIONAL, STUDIO, GROWTH];

export const PLANS_HEADER: PlanHeaders = {
  [STARTUP]: {
    key: STARTUP,
    name: 'Startup',
    price: 'Free',
    description: 'Create a profile and get exposure for your business services',
    icon: StartUpIcon,
  },
  [PROFESSIONAL]: {
    key: PROFESSIONAL,
    name: 'Professional',
    price: '$100/year',
    description: 'Create a profile and get exposure for your business services',
    icon: ProfessionalIcon,
  },
  [STUDIO]: {
    key: STUDIO,
    name: 'Studio',
    price: '$400/year',
    description: 'Create a profile and get exposure for your business services',
    icon: StudioIcon,
  },
  [GROWTH]: {
    key: GROWTH,
    name: 'Growth',
    price: '$900/year',
    description: 'Create a profile and get exposure for your business services',
    icon: GrowthIcon,
  },
};

export const SERVICES: Services = {
  [SERVICES_SHOWCASE]: {
    serviceKey: SERVICES_SHOWCASE,
    name: 'Services Showcase',
    isHeader: true,
    type: TEXT,
    theme: WHITE,
  },
  [SERVICE_LISTING]: {
    serviceKey: SERVICE_LISTING,
    name: 'Service Listing',
    type: TEXT,
    theme: GRAY,
  },
  [VERIFIED_PROFILE]: {
    serviceKey: VERIFIED_PROFILE,
    name: 'Verified Profile',
    type: ICON,
    theme: WHITE,
  },
  [FEATURED_PROFILE]: {
    serviceKey: FEATURED_PROFILE,
    name: 'Featured Profile & Services',
    type: ICON,
    theme: GRAY,
  },

  // Business Development
  [BUSINESS_DEVELOPMENT]: {
    serviceKey: BUSINESS_DEVELOPMENT,
    name: 'Business Development',
    isHeader: true,
    type: TEXT,
    theme: WHITE,
  },
  [AUTHENTICATION_VERIFICATION_SUPPORT]: {
    serviceKey: AUTHENTICATION_VERIFICATION_SUPPORT,
    name: 'Authentication & verification support',
    type: ICON,
    theme: GRAY,
  },
  [VERIFIED_CLIENT_REVIEWS]: {
    serviceKey: VERIFIED_CLIENT_REVIEWS,
    name: 'Verified Client Reviews',
    type: ICON,
    theme: WHITE,
  },
  [CLIENT_REVIEW_WITH_FOLLOW_UP]: {
    serviceKey: CLIENT_REVIEW_WITH_FOLLOW_UP,
    name: 'Client review with follow up',
    type: ICON,
    theme: GRAY,
  },
  [RECEIVE_RESPOND_INQUIRIES]: {
    serviceKey: RECEIVE_RESPOND_INQUIRIES,
    name: 'Receive & respond to inquiries',
    type: ICON,
    theme: WHITE,
  },
  [SUB_ACCOUNTS]: {
    serviceKey: SUB_ACCOUNTS,
    name: 'Sub accounts',
    type: TEXT,
    theme: GRAY,
  },

  // Customer service
  [CUSTOMER_SERVICE]: {
    serviceKey: CUSTOMER_SERVICE,
    name: 'Customer service',
    isHeader: true,
    type: TEXT,
    theme: WHITE,
  },
  [PERSONALISED_CUSTOMER_SERVICE]: {
    serviceKey: PERSONALISED_CUSTOMER_SERVICE,
    name: 'Personalised customer service',
    type: TEXT,
    theme: GRAY,
  },
  [LEGAL_ADVISORY_SERVICE]: {
    serviceKey: LEGAL_ADVISORY_SERVICE,
    name: 'Legal Advisory service',
    type: ICON,
    theme: WHITE,
  },
  [PRODUCT_PROFILE_ADVISORY_SERVICE]: {
    serviceKey: PRODUCT_PROFILE_ADVISORY_SERVICE,
    name: 'Product/Profile Advisory service',
    type: ICON,
    theme: GRAY,
  },
};

// services order
export const SERVICE_ROWS_ORDER: ServiceRowOrder[] = [
  // Services Showcase section
  SERVICES.SERVICES_SHOWCASE,
  SERVICES.SERVICE_LISTING,
  SERVICES.VERIFIED_PROFILE,
  SERVICES.FEATURED_PROFILE,

  // Business Development
  SERVICES.BUSINESS_DEVELOPMENT,
  SERVICES.AUTHENTICATION_VERIFICATION_SUPPORT,
  SERVICES.VERIFIED_CLIENT_REVIEWS,
  SERVICES.CLIENT_REVIEW_WITH_FOLLOW_UP,
  SERVICES.RECEIVE_RESPOND_INQUIRIES,
  SERVICES.SUB_ACCOUNTS,

  // Customer service
  SERVICES.CUSTOMER_SERVICE,
  SERVICES.PERSONALISED_CUSTOMER_SERVICE,
  SERVICES.LEGAL_ADVISORY_SERVICE,
  SERVICES.PRODUCT_PROFILE_ADVISORY_SERVICE,
];

// mobile Service order
export const MOBILE_SERVICES_ORDER: MobileService[] = [
  {
    ...SERVICES.SERVICES_SHOWCASE,
    subServices: [
      SERVICES.SERVICE_LISTING,
      SERVICES.VERIFIED_PROFILE,
      SERVICES.FEATURED_PROFILE,
    ],
  },
  {
    ...SERVICES.BUSINESS_DEVELOPMENT,
    subServices: [
      SERVICES.AUTHENTICATION_VERIFICATION_SUPPORT,
      SERVICES.VERIFIED_CLIENT_REVIEWS,
      SERVICES.CLIENT_REVIEW_WITH_FOLLOW_UP,
      SERVICES.RECEIVE_RESPOND_INQUIRIES,
      SERVICES.SUB_ACCOUNTS,
    ],
  },
  {
    ...SERVICES.CUSTOMER_SERVICE,
    subServices: [
      SERVICES.PERSONALISED_CUSTOMER_SERVICE,
      SERVICES.LEGAL_ADVISORY_SERVICE,
      SERVICES.PRODUCT_PROFILE_ADVISORY_SERVICE,
    ],
  },
];

export const PLAN_SERVICES: PlanServices = {
  [STARTUP]: {
    [SERVICES_SHOWCASE]: null,
    [SERVICE_LISTING]: '1',
    [VERIFIED_PROFILE]: '',
    [FEATURED_PROFILE]: null,
    [BUSINESS_DEVELOPMENT]: null,
    [AUTHENTICATION_VERIFICATION_SUPPORT]: null,
    [VERIFIED_CLIENT_REVIEWS]: null,
    [CLIENT_REVIEW_WITH_FOLLOW_UP]: null,
    [RECEIVE_RESPOND_INQUIRIES]: Checked,
    [SUB_ACCOUNTS]: '',
    [CUSTOMER_SERVICE]: null,
    [PERSONALISED_CUSTOMER_SERVICE]: 'Email support',
    [LEGAL_ADVISORY_SERVICE]: null,
    [PRODUCT_PROFILE_ADVISORY_SERVICE]: null,
  },
  [PROFESSIONAL]: {
    [SERVICES_SHOWCASE]: null,
    [SERVICE_LISTING]: '3',
    [VERIFIED_PROFILE]: Checked,
    [FEATURED_PROFILE]: null,
    [BUSINESS_DEVELOPMENT]: null,
    [AUTHENTICATION_VERIFICATION_SUPPORT]: Checked,
    [VERIFIED_CLIENT_REVIEWS]: Checked,
    [CLIENT_REVIEW_WITH_FOLLOW_UP]: null,
    [RECEIVE_RESPOND_INQUIRIES]: Checked,
    [SUB_ACCOUNTS]: '1',
    [CUSTOMER_SERVICE]: null,
    [PERSONALISED_CUSTOMER_SERVICE]: 'Email support',
    [LEGAL_ADVISORY_SERVICE]: null,
    [PRODUCT_PROFILE_ADVISORY_SERVICE]: null,
  },
  [STUDIO]: {
    [SERVICES_SHOWCASE]: null,
    [SERVICE_LISTING]: '10',
    [VERIFIED_PROFILE]: Checked,
    [FEATURED_PROFILE]: Checked,
    [BUSINESS_DEVELOPMENT]: null,
    [AUTHENTICATION_VERIFICATION_SUPPORT]: Checked,
    [VERIFIED_CLIENT_REVIEWS]: Checked,
    [CLIENT_REVIEW_WITH_FOLLOW_UP]: null,
    [RECEIVE_RESPOND_INQUIRIES]: Checked,
    [SUB_ACCOUNTS]: '3',
    [CUSTOMER_SERVICE]: null,
    [PERSONALISED_CUSTOMER_SERVICE]: 'Chat support',
    [LEGAL_ADVISORY_SERVICE]: Checked,
    [PRODUCT_PROFILE_ADVISORY_SERVICE]: Checked,
  },
  [GROWTH]: {
    [SERVICES_SHOWCASE]: null,
    [SERVICE_LISTING]: '10',
    [VERIFIED_PROFILE]: Checked,
    [FEATURED_PROFILE]: Checked,
    [BUSINESS_DEVELOPMENT]: null,
    [AUTHENTICATION_VERIFICATION_SUPPORT]: Checked,
    [VERIFIED_CLIENT_REVIEWS]: Checked,
    [CLIENT_REVIEW_WITH_FOLLOW_UP]: Checked,
    [RECEIVE_RESPOND_INQUIRIES]: Checked,
    [SUB_ACCOUNTS]: '10',
    [CUSTOMER_SERVICE]: null,
    [PERSONALISED_CUSTOMER_SERVICE]:
      'Dedicated account manager\n + Chat support',
    [LEGAL_ADVISORY_SERVICE]: Checked,
    [PRODUCT_PROFILE_ADVISORY_SERVICE]: Checked,
  },
};

export const PLAN_OPTIONS: PlansOption[] = [
  { value: STARTUP, label: 'Startup' },
  { value: PROFESSIONAL, label: 'Professional' },
  { value: STUDIO, label: 'Studio' },
  { value: GROWTH, label: 'Growth' },
];
