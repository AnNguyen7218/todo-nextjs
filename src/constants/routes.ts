export const ROUTES = {
  HOME: '/',
  SIGN_UP: '/sign-up',
  SIGN_IN: '/sign-in',
  EMAIL_CONFIRMATION: '/email-confirmation',
  FORGOT_PASSWORD: '/forgot-password',
  CHANGE_PASSWORD: '/change-password',
  GETTING_STARTED: '/getting-started',
  EMAIL_CONFIRMATION_SUCCESS: '/email-confirmation-success',
  EMAIL_CONFIRMATION_FAILURE: '/email-confirmation-failure',
  PROFILE: '/profile',
};
