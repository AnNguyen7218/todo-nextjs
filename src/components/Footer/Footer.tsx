import React from 'react';
import cx from 'classnames';
import { useRouter } from 'next/router';
import { ROUTES } from 'constants/routes';
import { FOOTER_LINKS_ICON, FOOTER_LINKS_TEXT } from 'constants/footer';
import { Logo, Container, HyperLink } from 'components';
import css from './Footer.scss';

const getFooterClassName = (pathname: string) => {
  switch (pathname) {
    case ROUTES.EMAIL_CONFIRMATION:
    case ROUTES.EMAIL_CONFIRMATION_SUCCESS:
    case ROUTES.EMAIL_CONFIRMATION_FAILURE:
    case ROUTES.SIGN_UP:
    case ROUTES.SIGN_IN:
    case ROUTES.GETTING_STARTED:
    case ROUTES.FORGOT_PASSWORD:
    case ROUTES.CHANGE_PASSWORD:
      return css['auto-hidden'];
  }
};

const Footer = () => {
  const router = useRouter();
  const className = getFooterClassName(router.pathname);
  return (
    <div className={cx(css['wrapper'], className)}>
      <Container className={css['container']}>
        <div className={css['top-section']}>
          <Logo theme="light" />
          <div className={css['icons-wrapper']}>
            {FOOTER_LINKS_ICON.map((link, index) => {
              const { url, icon: Icon } = link;
              return (
                <HyperLink href={url} key={index} className={css['link']}>
                  <Icon className={css['icon']} />
                </HyperLink>
              );
            })}
          </div>
        </div>
        <div className={css['bottom-section']}>
          <p className={css['sub-title']}>Â© 2020 Intrepid</p>
          <div className={css['texts-wrapper']}>
            {FOOTER_LINKS_TEXT.map((link, index) => {
              const { url, title } = link;
              return (
                <HyperLink className={css['link']} key={index} href={url}>
                  <span className={css['link-text']}>{title}</span>
                </HyperLink>
              );
            })}
          </div>
        </div>
      </Container>
    </div>
  );
};
export default Footer;
