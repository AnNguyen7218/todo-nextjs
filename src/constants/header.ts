// import { logout } from 'utils/auth';
import { ROUTES } from 'constants/routes';

export const MENU_ITEMS: MenuItemType[] = [
  {
    title: 'item1',
  },
  {
    title: 'item2',
  },
  {
    title: 'item3',
  },
  {
    title: 'item4',
  },
  {
    title: 'item5',
  },
];

export const USER_MENU: UserMenuType[] = [
  {
    title: 'Your profile',
    type: 'link',
    href: ROUTES.PROFILE,
  },
  {
    title: 'Log out',
    type: 'action',
    // action: logout,
  },
];
