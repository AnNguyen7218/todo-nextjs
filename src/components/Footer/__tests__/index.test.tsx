import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { ROUTES } from 'constants/routes';
import Footer from '../Footer';

afterEach(cleanup);

describe('Footer Component', () => {
  test('Footer should match snapshot', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.HOME,
    }));
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  test('Footer should match snapshot when current route is signup page', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.SIGN_UP,
    }));
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
  test('Footer should match snapshot when current route is confirm email page', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.EMAIL_CONFIRMATION,
    }));
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
  test('Footer should match snapshot when current route is login page', () => {
    jest.spyOn(require('next/router'), 'useRouter').mockImplementation(() => ({
      pathname: ROUTES.SIGN_IN,
    }));
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
