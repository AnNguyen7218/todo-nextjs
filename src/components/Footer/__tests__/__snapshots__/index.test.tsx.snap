// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Footer should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          © 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer Component Footer should match snapshot when current route is confirm email page 1`] = `
<div>
  <div
    class="wrapper auto-hidden"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          © 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer Component Footer should match snapshot when current route is login page 1`] = `
<div>
  <div
    class="wrapper auto-hidden"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          © 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Footer Component Footer should match snapshot when current route is signup page 1`] = `
<div>
  <div
    class="wrapper auto-hidden"
  >
    <div
      class="container container"
    >
      <div
        class="top-section"
      >
        <ic-logo.svg
          classname="logo--light"
          height="27"
          width="100"
        />
        <div
          class="icons-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-facebook.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-insta.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-twitter.svg
              classname="icon"
            />
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <ic-linkin.svg
              classname="icon"
            />
          </a>
        </div>
      </div>
      <div
        class="bottom-section"
      >
        <p
          class="sub-title"
        >
          © 2020 Intrepid
        </p>
        <div
          class="texts-wrapper"
        >
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Term
            </span>
          </a>
          <a
            class="link link--white link--hover link"
            href="/#"
          >
            <span
              class="link-text"
            >
              Privacy
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
